sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -604,7 +604,13 @@ class Analyzer(
 
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
       case i @ InsertIntoTable(u: UnresolvedRelation, parts, child, _, _) if child.resolved =>
-        i.copy(table = EliminateSubqueryAliases(lookupTableFromCatalog(u)))
+        val newTable = EliminateSubqueryAliases(lookupTableFromCatalog(u))
+        // Inserting into a view is not allowed, we should throw an AnalysisException.
+        if (newTable.isInstanceOf[View]) {
```

Comment:
Discussed with @cloud-fan and the concern is that in case the child of the view node is invalid(e.g. exists cyclic view reference or exceed max reference depth), we should still indicate that INSERT INTO VIEW is not allowed(instead of other error messages), so we should not resolve the child of the view, instead we throw an Exception here.